### PR TITLE
Implement `tolowercase`

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Towards: https://github.com/lfortran/lfortran/issues/3550

### CODE REVIEW

File: src/libasr/pass/intrinsic_functions.h [hunk 1]
```diff
@@ -2425,6 +2426,73 @@ namespace Leadz {
...
+        ASR::expr_t* ichar_Z = ASRUtils::EXPR(ASR::make_Ichar_t(al, loc,
+                                    ASRUtils::EXPR(ASR::make_StringConstant_t(al, loc,
+                                                s2c(al, "Z"), arg_types[0])), int32, nullptr));
+        ASR::expr_t* ichar_a = ASRUtils::EXPR(ASR::make_Ichar_t(al, loc,
+                                    ASRUtils::EXPR(ASR::make_StringConstant_t(al, loc,
+                                                s2c(al, "a"), arg_types[0])), int32, nullptr));
+
+        body.push_back(al, b.While(cond, {
+            b.If(And(iGtE(ichar_res, ichar_A), iLtE(ichar_res, ichar_Z)), {
+                b.Assignment(result, iSub(iAdd(ichar_res, ichar_a), ichar_A))
```
Comment (HarshitaKalani) [inline src/libasr/pass/intrinsic_functions.h:2484]:
```suggestion
                b.Assignment(result, achar(iSub(iAdd(ichar_res, ichar_a), ichar_A)))
```
This needs intrinsic `achar`.

Comment (HarshitaKalani) [reply src/libasr/pass/intrinsic_functions.h:2484]: Opened an issue for this: https://github.com/lfortran/lfortran/issues/3557

File: src/libasr/pass/intrinsic_function_registry.h [hunk 2]
```diff
@@ -639,6 +644,7 @@ namespace IntrinsicElementalFunctionRegistry {
                 {"ior", {&Ior::create_Ior, &Ior::eval_Ior}},
                 {"ieor", {&Ieor::create_Ieor, &Ieor::eval_Ieor}},
                 {"leadz", {&Leadz::create_Leadz, &Leadz::eval_Leadz}},
+                {"tolowercase", {&ToLowerCase::create_ToLowerCase, &ToLowerCase::eval_ToLowerCase}},
```
Comment (certik) [inline src/libasr/pass/intrinsic_function_registry.h:647]: In order to test this function, we could name it `_lfortran_tolowercase` in the frontend (keep `ToLowerCase` in ASR), and just write a regular lfortran test for it.

File: src/libasr/asr_builder.h [hunk 3]
```diff
@@ -125,6 +125,7 @@ class ASRBuilder {
         type, nullptr))
     #define ListAppend(x, val) STMT(ASR::make_ListAppend_t(al, loc, x, val))
 
+    #define ch(x, t) ASRUtils::EXPR(ASR::make_StringConstant_t(al, loc, x, t))
```
Comment (Shaikh-Ubaid) [inline src/libasr/asr_builder.h:128]: There is already a `StringConstant` definition on line `133`. Why do we need this again?

Comment (HarshitaKalani) [reply src/libasr/asr_builder.h:128]: I was actually trying to create a function which can convert ascii values to char. `StringConstant` didn't help, so was trying some other approach. This is not needed, will remove this.

File: integration_tests/intrinsics_171.f90 [hunk 4]
```diff
@@ -0,0 +1,5 @@
+program main
```
[the remark this answers is not in the document]
Comment (HarshitaKalani) [reply integration_tests/intrinsics_171.f90:1]: I'll correct this

File: integration_tests/intrinsics_171.f90 [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+    character(5):: name = "hELLo"
+    print*, _lfortran_tolowercase("HeLo#")
+    print*, _lfortran_tolowercase(name)
```
[the remark this answers is not in the document]
Comment (certik) [reply integration_tests/intrinsics_171.f90:4]: It has to be debugged.

Also add a check using `if` and `error stop` to ensure it is computing correctly.

This design is good for now, later we can improve upon it (see https://github.com/lfortran/lfortran/issues/3582).

**Comment by HarshitaKalani** [comment]
Ready for review!

**Review by certik — APPROVED** [review]
I think this looks good.

**Pull request merged**